lxd/storage_volumes.go

```diff
@@ -81,6 +81,53 @@ var storagePoolVolumeTypeCmd = APIEndpoint{
 	Put:    APIEndpointAction{Handler: storagePoolVolumePut, AccessHandler: allowPermission(entity.TypeStorageVolume, auth.EntitlementCanEdit, "poolName", "type", "volumeName")},
 }
 
+// storagePoolVolumeTypeAccessHandler returns an access handler which checks the given entitlement on a storage volume.
+func storagePoolVolumeTypeAccessHandler(entitlement auth.Entitlement) func(d *Daemon, r *http.Request) response.Response {
+	return func(d *Daemon, r *http.Request) response.Response {
+		s := d.State()
+		err := addStoragePoolVolumeDetailsToRequestContext(s, r)
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		details, err := request.GetCtxValue[storageVolumeDetails](r.Context(), ctxStorageVolumeDetails)
+		if err != nil {
+			return response.SmartError(err)
+		}
+
+		var target string
+
+		// Regardless of whether the caller specified a target parameter, we do not add it to the authorization check if
+		// the storage pool is remote. This is because the volume in the database has a NULL `node_id`, so the URL uniquely
+		// identifies the volume without the target parameter.
+		if !details.pool.Driver().Info().Remote {
+			// If the storage pool is local, we need to add a target parameter to the authorization check URL for the
+			// auth subsystem to consider it unique.
+
+			// If the target parameter was specified, use that.
+			target = request.QueryParam(r, "target")
```

Comment:
I'm a bit confused about this part.

On the one hand, you're added target to the storage volume context field in `addStoragePoolVolumeDetailsToRequestContext` , so it would seem this should be taken out of the context rather than the URL.

Also I suggested earlier that as target isn't specific to storage volumes that it shouldn't be in the storage volume context field and could be its own dedicated context field, but if we are just able to access it via the URL here, why do we need it in the context at all (in any field)?

Reply:
I think on balance from your comments I will:
1. Remove `requestProject` from `storage{Volume,Bucket}Details`. We can just use `request.ProjectParam` for this. I don't think we need to dedicate a CtxKey for this since it is easy to parse without additional error handling.
2. Remove `effectiveProject` from `storage{Volume,Bucket}Details`. We will expect this to be available in the context via `request.CtxEffectiveProjectName`.
3. Remove `target` from `storage{Volume,Bucket}Details`. We can just use `request.QueryParam(r, "target")` for this. I don't think we need to dedicate a CtxKey for this since it is easy to parse without additional error handling.

Reply:
sounds good!

Reply:
I've updated the PR with these changes now.